lib/socket.js

```diff
@@ -99,6 +99,19 @@ module.exports = class UDXSocket extends events.EventEmitter {
     this.emit('busy')
   }
 
+  _addStream (stream) {
+    if (this.idle) this._onbusy()
+
+    this.streams.add(stream)
+  }
+
+  _removeStream (stream) {
+    this.streams.delete(stream)
+    this._closeMaybe()
+
+    if (this.idle) this._onidle()
```

Comment:
@mafintosh Should this actually happen before `_closeMaybe()`?

Reply:
yea, i think that makes sense actually.